core/jazz_slack-channel/test/test.js

```diff
@@ -1,11 +1,1025 @@
+        sinon.assert.calledOnce(reqStub);
+
+        getToken.restore();
+        reqStub.restore();
+      });
+    });
+
+    it("should indicate error if send email API request fails", () => {
+      let getToken = sinon.stub(index, 'getToken').resolves("authToken");
+      reqStub = sinon.stub(request, "Request").callsFake(obj => {
+        return obj.callback(err, null, null);
+      })
+
+      index.notifyUser(config, slackChannelMembers, channelInfo)
+      .catch(error => {
+        expect(error).to.include(err);
+
+        sinon.assert.calledOnce(getToken);
+        sinon.assert.calledOnce(reqStub);
+
+        getToken.restore();
+        reqStub.restore();
+      });
+    });
+
+    it("should indicate error if getToken fails", () => {
+      let getToken = sinon.stub(index, 'getToken').rejects(err);
+
+      index.notifyUser(config, slackChannelMembers, channelInfo)
+      .catch(error => {
+        expect(error).to.include(err);
+        sinon.assert.calledOnce(getToken);
+        getToken.restore();
+      });
+    });
+
+  });
+
+  describe('handler', () => {
+    it("should successufully execute handler function", () => {
+      let resObj = {
+        data: {
+          channelInfo: {
+            id: "CX123456",
+            name: event.body.channel_name,
+            link: config.slack_url
+          },
+          slackChannelMembers: event.body.users
+        }
+      }
+      let genericInputValidation = sinon.stub(index, 'genericInputValidation').resolves();
+      let createPublicSlackChannel = sinon.stub(index, 'createPublicSlackChannel').resolves("channelInfo");
+      let addMemberToChannel = sinon.stub(index, 'addMemberToChannel').resolves(resObj);
+
+      index.handler(event, context, (err, res) => {
+        expect(res).to.have.all.keys('data', 'input');
+        expect(res.data).to.have.all.keys('channel_id', 'channel_name', 'channel_link', 'members');
+        expect(res.data.channel_id).to.eq(resObj.data.channelInfo.id);
+        expect(res.data.channel_name).to.eq(resObj.data.channelInfo.name);
+        expect(res.data.channel_link).to.eq(resObj.data.channelInfo.link);
+        expect(res.data.members).to.eq(resObj.data.slackChannelMembers);
+
+        sinon.assert.calledOnce(genericInputValidation);
+        sinon.assert.calledOnce(createPublicSlackChannel);
+        sinon.assert.calledOnce(addMemberToChannel);
+
+        genericInputValidation.restore();
+        createPublicSlackChannel.restore();
+        addMemberToChannel.restore();
+      });
+    });
+
+    it("should indicate input validation error/ BadRequest", () => {
+      let inputError = {
+        result: "inputError",
+        message: "Provided invalid input"
+      };
+      let genericInputValidation = sinon.stub(index, 'genericInputValidation').rejects(inputError);
+
+      index.handler(event, context, (err, res) => {
+        expect(err).to.include('{"errorType":"BadRequest","message":"'+inputError.message+'"}');
```

Comment:
give spaces ("' + '" )
fix in other places as well

Reply:
Done